service/src/main/java/bio/terra/catalog/rawls/RawlsService.java

```diff
@@ -0,0 +1,55 @@
+package bio.terra.catalog.rawls;
+
+import bio.terra.catalog.config.RawlsConfiguration;
+import bio.terra.catalog.model.SystemStatusSystems;
+import bio.terra.rawls.api.StatusApi;
+import bio.terra.rawls.client.ApiClient;
+import bio.terra.rawls.client.ApiException;
+import javax.ws.rs.client.Client;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class RawlsService {
+  private static final Logger logger = LoggerFactory.getLogger(RawlsService.class);
+
+  private final RawlsConfiguration rawlsConfig;
+  private final Client commonHttpClient;
+
+  @Autowired
+  public RawlsService(RawlsConfiguration rawlsConfig) {
+    this.rawlsConfig = rawlsConfig;
+    this.commonHttpClient = new ApiClient().getHttpClient();
+  }
+
+  private ApiClient getApiClient() {
+    // Share one api client across requests.
+    return new ApiClient().setHttpClient(commonHttpClient).setBasePath(rawlsConfig.basePath());
+  }
+
+  public boolean statusIsOk() throws ApiException {
+    new StatusApi(getApiClient()).systemStatus();
+    return true;
+  }
+
+  public SystemStatusSystems status() {
+    var result = new SystemStatusSystems();
+    try {
```

Comment:
I am confused by this try/catch block, but it is probably just my Java understanding.

If the result is not ok, or it is thrown we log the error and add the message to the result - when is the error thrown vs gettting a false on `statusIsOk`?

Could this be simplified?

Reply:
There are some issues with the Rawls status endpoint that may need to be resolved in the future. As a result, although the Rawls method currently returns nothing or throws, it is possible that `statusIsOk` returns true, false, or throws in the future. That's why I have a condition that checks the result and that also check if the method throws.